### DISCUSSION TITLE
Disjunctions for Vehicle Routing Problem with Time Windows

### DISCUSSION THREAD

**Discussion by rahulj2409** [google/or-tools#2664] [Routing (and legacy CP) questions]
Python
Can you have the penalties for dropping nodes VIA add_disjunction in the scenario of VRPTW. The case is that I have a set amount of drivers and I want to visit as many nodes (accounting for time window constraints). Would adding a penalty via addDisjunction(node, penalty) allow it to select as many nodes to get the least total penalty? Furthermore, is there a way to do this as well but also adding the time_seconds constraint IE (only run for 60 seconds).

Thanks!

**Comment by Mizux** [comment]
1) if penalty is higher than the increase in the arc cost involved to visit the node, then solver would have incentive to visit the node instead of dropping it.
2) Yes, solver goal is always to minimize the objective which is the sum of all costs...
3) If "only run for 60" means get an answer after 60s of search then set search parameter time limit, this is unrelated to disjunction stuff...

**Reply by rahulj2409** [reply]
I made the penalty as low as possible and it still ran for a super long time for 100 nodes and 10 vehicles. Shouldn't the whole point be that allowing the vehicles to drop nodes make it run faster because it doesn't need to visit all the nodes. My point with 3. is that is it possible to make it like "try to visit as many nodes within 60 seconds while minimizing the total penalty".

Thanks!
